### PR TITLE
Add univariate list, contrib page & Travis update.

### PR DESCRIPTION
**Pull request by taqtiqa-mark**
Update to Travis staged build.
 - The exisitng build has these drawbacks:
    - Code in the after_success: part of the build does not fail the
      build.  This makes it difficult to verify that (i) the doc build
      works and (ii) that doctest etc. still passes.
    - Doc building runs in the global environment, and is thus affected
      by the surroundings.

Add univariate listing implemented, not and won't.
 - Closes issue JuliaStats/Distributions.jl #124 and issue
   JuliaStats/Distributions.jl #668.
 - Includes (most) Wikipedia list of continuous univariate dist.

Add Contributing page to docs site.
 - Modelled on Julia and Documenter.jl page.
 - Included excerpt from README.md.

### CODE REVIEW

File: docs/src/contributing.md [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+3. At the end of the tests, `Test.detect_ambiguities(Distributions)` is run to check method ambiguities. Verify that your modified code did not yield method ambiguities.
+4. Make according modifications to the `docs` folder, build the documentation locally with `$ julia docs/make.jl`, verify that your modifications display correctly and did not yield warnings.
+
+## Branches
+
+This package uses `release-*` branches for tagged minor versions. This follows the same approach used in the main Julia repository, albeit on a much more modest scale.
+
+Please open pull requests against the `master` branch rather than any of the `release-*` branches whenever possible.
+
+### Backports
```
Comment (matbesancon) [inline docs/src/contributing.md]: Don't think we backport things in Distributions

File: docs/src/contributing.md [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+Bug fixes are backported to the `release-*` branches using `git cherry-pick -x` and will become available in point releases of that particular minor version of the package.
+
+Feel free to nominate commits that should be backported by opening an issue. Requests for new point releases to be tagged in `METADATA.jl` can also be made in the same way.
+
+### `release-*` branches
```
Comment (matbesancon) [inline docs/src/contributing.md]: We don't have release branches

File: docs/src/univariate-list.md [hunk 3]
```diff
@@ -0,0 +1,269 @@
+# [Univariate Distributions List](@id univariates-list)
```
Comment (matbesancon) [inline docs/src/univariate-list.md]: Is is not redundant with `univariates.md`? Plus it creates a hard to track distributions list that needs to be updated. The issue tracker might be more adapted

**Comment by matbesancon** [comment]
thanks for this PR @taqtiqa-mark. There has been some changes on Travis on the last PRs, would you mind merging master into your branch? Also, one different build stage has been defined for docs in the latest commits.
I left some other comments in the code for the other parts

File: docs/src/contributing.md (hunk 2 above)
Comment (taqtiqa-mark) [reply docs/src/contributing.md]: Hmm, on reflection and further reading: 
What to make of the `/^v\d+\.\d+(\.\d+)?(-\S*)?$/` filter that was just added to Travis? Should this section be reworded to refer to the Travis regex naming convention? Or removed entirely?

File: docs/src/univariate-list.md (hunk 3 above)
Comment (taqtiqa-mark) [reply docs/src/univariate-list.md]: Re redundancy:  
 - Think of this as an index page, with all the functionality and convenience use cases that index pages were created for.
 - Also think of this as a cross reference tool - not all distributions are known by the same name across all SME domains.

Happy to stand corrected on the cross referencing.

Yes this distributions list would need to be updated as distributions are added - more importantly reviews would be encouraged to identify synonyms that should be cross referenced to the incoming distribution.

I think the current issues are only in the search path of devs, not general users.

File: docs/src/contributing.md (hunk 1 above)
Comment (taqtiqa-mark) [reply docs/src/contributing.md]: So that section would be replaced by the 1-liner:  Changes are not backported in Distributions.jl

Comment (matbesancon) [reply docs/src/contributing.md]: Yes, or even not mention it. It's not expected for most Julia packages. Even less while Distributions is < 1.0

File: docs/src/contributing.md (hunk 2 above)
Comment (matbesancon) [reply docs/src/contributing.md]: It can be removed. The travis regex is just declaring to push to documentation major and minor releases

**Comment by taqtiqa-mark** [comment]
@matbesancon is there a reason you picked out only the documentation job and kept the matrix instructions.  

Ack. Travis-CI is fast moving and the doc's haven't always been able to keep up.... My understanding is the the way `jobs` are used in `.travis.yml` in this PR is the updated approach.  Mixing matrix and jobs has led to some subtle pain points previously.  

Is there a reason you want to retain the superseded syntax and behavior?

**Comment by matbesancon** [comment]
I copied the travis template from:
https://github.com/fredrikekre/Literate.jl/blob/master/.travis.yml
Agreed this is more of a hack, I'm fine with changing to jobs completely.

**Comment by matbesancon** [comment]
Again, I'm not sold on the index page, documentation is a high burden to maintain, as shown by the huge recent PR on them. I prefer an issue for the implemented and non implemented ones, like #124, to which we can point in the README

**Comment by taqtiqa-mark** [comment]
No worries.  You know where to find it if you change you mind.

**Comment by matbesancon** [comment]
thanks for understanding

File: docs/make.jl [hunk 4]
```diff
@@ -10,12 +10,14 @@ makedocs(
         "starting.md",
         "types.md",
         "univariate.md",
+        "univariate-list.md",
```
Comment (matbesancon) [inline docs/make.jl]:
```suggestion
```

**Comment by matbesancon** [comment]
One last suggestion in the diff, and think of merging master into your branch to check that you're up to date

File: docs/make.jl [hunk 5]
```diff
@@ -16,6 +16,7 @@ makedocs(
         "mixture.md",
         "fit.md",
         "extends.md",
+        "contributing.md"
```
Comment (matbesancon) [inline docs/make.jl]:
```suggestion
        "contributing.md",
```
So that adding new lines reduces the diff

File: docs/src/contributing.md [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+
+1. If you need explanation on how to do X, Y using Distributions,
+feel free to ask on the Julia [Discourse](https://discourse.julialang.org/c/domain/stats)
+or [Slack](https://julialang.slack.com), get an invitation
+[here](https://slackinvite.julialang.org/).
+
+2. If you have a bug linked with Distributions, check that it has
+not been reported yet on the issues of the repository.
+If not, you can file a new issue, add your version of the package
+which you can get with this command in the Julia REPL:
```
Comment (matbesancon) [inline docs/src/contributing.md]:
```suggestion
which you can obtain with this command in the Julia REPL:
```

File: docs/src/contributing.md [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+### Requirements
+
+Distributions is a central package which many rely on,
+the following are required for contributions to be accepted:
+1. Docstrings must be added to all interface and non-trivial functions.
+2. Tests validating the modified behavior in the `test` folder. If new test files are added, do not forget to add them in `test/runtests.jl`. Cover possible edge cases. Run the tests locally before submitting the PR.
+3. At the end of the tests, `Test.detect_ambiguities(Distributions)` is run to check method ambiguities. Verify that your modified code did not yield method ambiguities.
+4. Make according modifications to the `docs` folder, build the documentation locally with `$ julia docs/make.jl`, verify that your modifications display correctly and did not yield warnings.
+
+## Branches
```
Comment (matbesancon) [inline docs/src/contributing.md]: This section can be removed since no release branch

**Comment by matbesancon** [comment]
@taqtiqa-mark not sure the build is allowing nightly failure as-is, I relaunched in doubt

File: .travis.yml [hunk 8]
```diff
@@ -1,16 +1,21 @@
 language: julia
+
+if: branch=master
```
Comment (ararslan) [inline .travis.yml]: This doesn't seem right, as it disables branch builds. Doesn't it?

**Comment by ararslan** [comment]
Seems good to me overall except for the Travis changes. Ideally it should resemble that of other packages as closely as possible.

**Comment by matbesancon** [comment]
@ararslan I'm good to merge whenever you are

**Comment by ararslan** [comment]
I'd prefer if we didn't do the whole build stage thing. Ideally Travis setup files can be copied and pasted between repos with little to no editing required, and the vast majority of packages I've seen do not use stages this way, only for documentation.

**Comment by matbesancon** [comment]
I would be fine with this one, the resulting Travis page reads well. It's
not obvious why we should remain compliant with other packages, probably
some others will switch to build stages, now that it's done for
documentation.

If we want to make this package a bit more standard, I think tests would be
a more pressing starting point (switching from JSON/text files to Julia
data structures.

**Comment by mschauer** [comment]
What is the source/inspiration for the style guides?

**Comment by taqtiqa-mark** [comment]
>Add Contributing page to docs site.
>
>  -  Modelled on Julia and Documenter.jl page.
>  - Included excerpt from README.md.

**Comment by mschauer** [comment]
Ah, I'll have a look. As a side remark, this should have been perhaps three independent PRs.

**Comment by mschauer** [comment]
@taqtiqa-mark Would you split this up into a travis update and separately a contrib page? Then we can go ahead with the travis update.

**Comment by matbesancon** [comment]
ping @taqtiqa-mark for the requested changes
